lib/Services/RetrieveMetadata.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Services;
+
+
+use OCP\IDBConnection;
+
+class RetrieveMetadata {
+
+	protected $dbConnection;
+
+	/**
+	 * @param IDBConnection $connection
+	 */
+	public function __contruct(IDBConnection $connection){
+		$this->dbConnection = $connection;
+	}
+
+	/**
+	 * @param $fileList : List of File IDs
+	 * @return array
+	 */
+	public function retrieve($fileList) {
+
+		$retrievedData = array();
+		$mapper = new ImageDimensionMapper($this->dbConnection);
+
+		$logger = \OC::$server->getLogger();
+
+		foreach ($fileList as $fileID) {
+			$entity = $mapper->find($fileID);
```

Comment:
Create the mapper
https://github.com/nextcloud/news/blob/7a3a22bd2757e129c8964d9ddcde15ec53b736bb/tests/Unit/Db/ItemMapperTest.php#L34

and then create the mocks like that?
https://github.com/nextcloud/news/blob/7a3a22bd2757e129c8964d9ddcde15ec53b736bb/tests/Unit/Db/ItemMapperTest.php#L495-L496